lib/templateFactory.js

```diff
@@ -80,33 +81,72 @@ class TemplateFactory extends BaseFactory {
     }
 
     /**
-     * Get a the latest template by config
+     * Get a the latest template by config using the full template name
      * @method getTemplate
-     * @param  {Object}     config               Config object
-     * @param  {String}     config.name          The template name
-     * @param  {String}     config.version       Version of the template
-     * @param  {String}     [config.label]       Label of template
-     * @return {Promise}                         Resolves template model or undefined if not found
+     * @param  {String}     fullTemplateName    Name of the template and the version or tag (e.g. chef/publish@1.2.3)
+     * @return {Promise}                        Resolves template model or undefined if not found
      */
-    getTemplate(config) {
-        // get all templates with the same name
-        return super.list({ params: { name: config.name } })
-            .then((templates) => {
-                // get templates that have version prefix as config.version
-                const filtered = templates.filter(template =>
-                    template.version.startsWith(config.version));
-
-                // Sort templates by descending order
-                filtered.sort((a, b) => compareVersions(b.version, a.version));
-
-                // If no config.label, return latest one
-                if (filtered.length > 0 && !config.label) {
-                    return filtered[0];
-                }
+    getTemplate(fullTemplateName) {
+        const match = TEMPLATE_NAME_REGEX.exec(fullTemplateName);
+        const templateName = match[TEMPLATE_NAME_MATCH];
+        const versionOrTag = match[VERSION_OR_TAG_MATCH];
+        const isVersion = VERSION_REGEX.exec(versionOrTag);
+
+        return new Promise((resolve) => {
+            // If no version or tag provided, return version is null
+            if (!versionOrTag) {
```

Comment:
Oh I see. You can keep it this way or you can do `if (versionOrTag && !isVersion)` on line 102 then you don't need these extra checks. Up to you.